DOC add link to plot_gpr_on_structured_data example in gaussian_process

ashbleu opened the pull request:
<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/scikit-learn/scikit-learn/blob/main/CONTRIBUTING.md
-->

#### Reference Issues/PRs
<!--
Example: Fixes #1234. See also #3456.
Please use keywords (e.g., Fixes) to create link to the issues or pull requests
you resolved, so that they will automatically be closed when your pull request
is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
-->

Towards #30621.


#### What does this implement/fix? Explain your changes.

Added link to the `plot_gpr_on_structured_data` example in the Gaussian Process Regression section of `gaussian_process` in the User Guide, since it isn't currently linked anywhere on that page.

#### Any other comments?


<!--
Please be aware that we are a loose team of volunteers so patience is
necessary; assistance handling other issues is very welcome. We value
all user contributions, no matter how minor they are. If we are slow to
review, either the pull request needs some benchmarking, tinkering,
convincing, etc. or more likely the reviewers are simply busy. In either
case, we ask for your understanding during the review process.
For more information, see our FAQ on this topic:
https://scikit-learn.org/dev/faq.html#why-is-my-pull-request-not-getting-any-attention.

Thanks for contributing!
-->

ashbleu committed:
DOC add link to plot_gpr_on_structured_data example in gaussian_process

StefanieSenger reviewed (commented):
Hello @ashbleu,

thank you for opening this PR.

You have identified that this example is hard to discover (currently there is no link to it anywhere in the docs). 

Instead of adding the link as a 5th link line in Examples, I would suggest to instead integrate it in the section about kernels in the user guide, here: https://scikit-learn.org/stable/modules/gaussian_process.html#gp-kernels.

- as a sentence 
- pointing out the example's demonstration on how to design a custom kernel for discrete data

Would you mind going back to make this change?

ashbleu committed:
Merge branch 'scikit-learn:main' into doc_gpr

ashbleu committed:
Move link to the GP kernels section

ashbleu commented:
Hi @StefanieSenger,

I have added a sentence about the example in the Kernels section, and removed it from the list of links in Examples. Please let me know if you have any other suggestions!

StefanieSenger reviewed (approved):
Thanks for your work, @ashbleu!

That looks very good. The sentence is in the right spot I would think and everything renders correctly. The CI failures are just http request errors and should go away if you re-trigger the CI by pushing an empty commit.

Would you like to take a look at this PR, @adrinjalali ?

ashbleu committed:
Merge branch 'main' into doc_gpr

adrinjalali committed:
Merge remote-tracking branch 'upstream/main' into doc_gpr

adrinjalali committed:
line length
